### DISCUSSION TITLE
Cant login.

### DISCUSSION THREAD

**Discussion by hammyexe** [Hubro/clipface#62] [Q&A] [answered]
Whenever I try to log in on clipface,I put in the password and it either infinitely loads, or I get this error message.

![Screenshot 2022-10-03 225501](https://user-images.githubusercontent.com/89997092/193667268-4f6524bc-71b6-4d5b-a50e-dc781ff0c9f1.png)

But whenever I try to log in without any passwords, it works fine.

**Comment by Hubro** [comment, marked as answer]
Please provide some details about your setup. Are you using HTTPS? Do you see any errors in the console when you try to log in? (Press <kbd>F12</kbd> in most browsers to show the console)

Do you see any errors on the server side when you try to log in? (If you're using Docker, use `docker logs <container name>` to get the logs.)

**Reply by hammyexe** [reply]
Sorry, I apologize. I was very tired because I was trying to fix this for a while and I didn't think of adding those details. I get no errors in my browser console, I have attached an image below of what my logs are.

![Screenshot 2022-10-04 143515](https://user-images.githubusercontent.com/89997092/193809977-5aa4b4ee-89fb-4596-a851-a9a3dd01dda0.png)

The server is hosted in my home on a linux ubuntu server. I don't know if this is important but I can only access the website with domain/public ip if im not connected to my home network or using a vpn. It works completely fine if I use anything other then my home connetion. I am able to login in the website and its https. But when I use the local ip 192.168.x.x, it is http and that is where i am getting the issue.

**Reply by Hubro** [reply]
Yes, you won't be able to use authentication without HTTPS if the `secure_cookies` option is set. This is documented in the README :slightly_smiling_face:

Basically, the `secure_cookies` option makes the browser refuse to send the authentication cookie to the server over regular HTTP since it's insecure.

If you really need to access Clipface using HTTP, change the `secure_cookies` setting. Though, be wary that your password and cookies will now be sent in plain text, which means they can trivially be snatched by anybody with access to your network traffic.

**Comment by hammyexe** [comment]
![Screenshot 2022-10-28 002224](https://user-images.githubusercontent.com/89997092/198400773-90c9ea4e-d627-439a-b062-d14c7ad86d82.png)


This is my config, it is turned off but I still get the error.

**Reply by Hubro** [reply]
My only guess then is that the cookie is cached as secure in the browser. To make sure, try using a different browser that you haven't opened Clipface with before, or use Incognito mode. Alternatively, if using Chrome, you can also go to dev tools (F12), go to the "Application" tab, then the "Storage" sub-tab, the click "Clear site data".

**Reply by hammyexe** [reply]
> My only guess then is that the cookie is cached as secure in the browser. To make sure, try using a different browser that you haven't opened Clipface with before, or use Incognito mode. Alternatively, if using Chrome, you can also go to dev tools (F12), go to the "Application" tab, then the "Storage" sub-tab, the click "Clear site data".

I just tried that, unfortunately I got the same error

**Reply by Hubro** [reply]
Then I suspect your settings are not active. Go to your Clipface instance, open developer tools (F12), navigate to "Application", open "Cookies" under "Storage" and expand your domain, find the cookie called "auth". Does it have the "Secure" column checked? If so, your settings are not active. See the README for instructions on how to apply your settings.

**Reply by hammyexe** [reply]
I don't see any cookies called auth

**Reply by Hubro** [reply]
Right, I suppose that could be the case if you're not visiting the page using HTTPS. Try recording your requests instead and check how Clipface sets the cookie.

1. Go to your clipface install
2. Open dev tools (F12)
3. Go to Network and check "Preserve log"
4. Go to the login page and log in
5. Look for the "login" request in the request log and click it
6. Under "Response Headers", look for "set-cookie"
7. Check if "set-cookie" includes the text "Secure;"

If it does, then Clipface is still setting the cookie as secure, meaning you can only authenticate when using HTTPS. In other words, your config is not active. Double check that your config file is mounted to the right place, double check the README. Also try setting `secure_cookies` to `false` using environment variables instead.

If authentication still doesn't work, it must be a bug in Clipface.
